python/src/aiconfig/scripts/aiconfig_cli.py

```diff
@@ -64,8 +65,33 @@ def _sigint(procs: list[subprocess.Popen[bytes]]) -> Result[str, str]:
         p.send_signal(signal.SIGINT)
     return Ok("Sent SIGINT to frontend servers.")
 
+def is_port_in_use(port: int) -> bool:
+    """ 
+    Checks if a port is in use at localhost.
+    
+    Creates a temporary connection.
+    Context manager will automatically close the connection
+    """
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(('localhost', port)) == 0
+
 
 def _run_editor_servers(edit_config: EditServerConfig) -> Result[list[str], str]:
+    
+    port = edit_config.server_port
+
+    while is_port_in_use(port):
+        LOGGER.warning(f"Port {port} is in use. Checking next port.")
+        port += 1
+
+    # Must reconstruct, EditServerConfig is an immutable type (frozen dataclass)
+    edit_config_dict = edit_config.model_dump()
+    edit_config_dict["server_port"] = port
+    edit_config = EditServerConfig(**edit_config_dict)
```

Comment:
`model_copy` is what you want here https://docs.pydantic.dev/latest/concepts/serialization/#model_copy